cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -187,16 +192,54 @@ func MustNew(cfg config.LocalConfig) *Daemon {
 	}
 }
 
-func Run(cfg config.LocalConfig, endpoint string) {
+func Run(cfg config.LocalConfig, endpoint string, adminEndpoint string) {
 	d := MustNew(cfg)
-	supporthttp.Run(supporthttp.Config{
-		ListenAddr: endpoint,
-		Handler:    d,
-		OnStarting: func() {
-			d.logger.Infof("Starting Soroban JSON RPC server on %v", endpoint)
-		},
-		OnStopped: func() {
-			d.Close()
-		},
-	})
+
+	server := &http.Server{
+		Addr:        endpoint,
+		Handler:     d,
+		ReadTimeout: defaultReadTimeout,
+	}
+
+	d.logger.Infof("Starting Soroban JSON RPC server on %v", endpoint)
+
+	go func() {
+		if err := server.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+			// Error starting or closing listener:
+			d.logger.Fatalf("Soroban JSON RPC server encountered fatal error: %v", err)
+			os.Exit(1)
+		}
+	}()
+	var adminServer *http.Server
+	if adminEndpoint != "" {
+		// after importing net/http/pprof, debug endpoints are implicitly registered in the default serve mux
+		adminServer = &http.Server{Addr: adminEndpoint, Handler: http.DefaultServeMux}
```

Comment:
G112: Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server (gosec)

Reply:
The admin port should never be exposed publicly. Way worse things can happen if it does.

Reply:
it wasn't me.. it's just to appease the linter.

Reply:
Ah, I see

Reply:
If you meant we should be adding a `//nolint:gosec` problem, I think it's better not to add noise to the file based on the tooling we are currently using.